### PR TITLE
Reworked ROS Interface

### PR DESCRIPTION
**Pull request by sasjonge**
This PR fixes the ROS Interface. It now builds with catkin and offers multiple interfaces when run with `roslaunch knowrob knowrob.launch`.

I renamed the project in package.xml and CMakelists from KnowRob to knowrob, because i got build warnings that the project name should be in lower case. 

The class InterfaceUtils.cpp have some common function other interfaces could also use, e.g. a rest interface.

It tested the 5 interfaces with a local integration test, at a later point we could have a look at an automatic testing of the interfaces via CI.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,9 +5,9 @@ if (POLICY CMP0148)
     cmake_policy(SET CMP0148 OLD)
 endif ()
 
-option(BUILD_ROS1_INTERFACE "Build the interface for ROS1 systems" OFF)
+option(BUILD_WITH_CATKIN "Build the interface for ROS1 systems" ON)
```
Comment (daniel86) [inline CMakeLists.txt]: Isn't catkin used by ROS2 too? I would vote calling the option "WITH_ROS1" or "WITH_ROS1_INTERFACE"

File: CMakeLists.txt [hunk 2]
```diff
@@ -115,18 +102,25 @@ if (false)
             EventToken.msg
     )
 
+    add_action_files(
+            DIRECTORY action
```
Comment (daniel86) [inline CMakeLists.txt]: maybe we should move actions also into a ros1 folder?

File: include/knowrob/integration/ros1/ROSInterface.h [hunk 3]
```diff
@@ -24,41 +22,93 @@
...
+        actionlib::SimpleActionServer <AskIncrementalAction> askincremental_action_server_;
+		actionlib::SimpleActionServer <AskIncrementalNextSolutionAction> askincremental_next_solution_action_server_;
+        actionlib::SimpleActionServer <TellAction> tell_action_server_;
+
+		// KnowledgeBase
         KnowledgeBase kb_;
-    public:
+
+		// Mutex to protect query_results_
+		std::mutex query_mutex_;
```
Comment (daniel86) [inline include/knowrob/integration/ros1/ROSInterface.h:50]: i think indentation is wrong here. maybe run auto-format on the changed files

File: include/knowrob/queries/QueryParser.h [hunk 4]
```diff
@@ -9,6 +9,7 @@
 #include "knowrob/formulas/Formula.h"
 #include "knowrob/formulas/Predicate.h"
 #include "knowrob/terms/Function.h"
+#include "boost/any.hpp"
```
Comment (daniel86) [inline include/knowrob/queries/QueryParser.h]: seems not needed, or move into source file if needed there

File: include/knowrob/integration/InterfaceUtils.h [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	 */
+	class InterfaceUtils {
+	public:
+		/**
+		 * Assert a list of statements.
+		 *
+		 * @param args the statements.
+		 * @return true if all statements are true.
+		 */
+		static bool assertStatements(const KnowledgeBase kb_, const std::vector<FormulaPtr> &args);
```
Comment (daniel86) [inline include/knowrob/integration/InterfaceUtils.h]: `kb_` is copied here, better pass in by reference

File: include/knowrob/integration/ros1/ROSInterface.h [hunk 6]
```diff
@@ -24,41 +22,93 @@
...
 
         // Action Servers
-        actionlib::SimpleActionServer <askallAction> askall_action_server_;
-        actionlib::SimpleActionServer <askoneAction> askone_action_server_;
-        actionlib::SimpleActionServer <askincrementalAction> askincremental_action_server_;
-        actionlib::SimpleActionServer <tellAction> tell_action_server_;
+        actionlib::SimpleActionServer <AskAllAction> askall_action_server_;
+        actionlib::SimpleActionServer <AskOneAction> askone_action_server_;
+        actionlib::SimpleActionServer <AskIncrementalAction> askincremental_action_server_;
+		actionlib::SimpleActionServer <AskIncrementalNextSolutionAction> askincremental_next_solution_action_server_;
```
Comment (daniel86) [inline include/knowrob/integration/ros1/ROSInterface.h]: intendation

File: src/integration/InterfaceUtils.cpp [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+InterfaceUtils::applyModality(const std::unordered_map<std::string, boost::any> &options,
+						   FormulaPtr phi) {
+	FormulaPtr mFormula = std::move(phi);
+
+	// Retrieve epistemicOperator and check if it is "BELIEF"
+	KB_INFO("[KnowRob] epistemicOperator.");
+	auto epistemicOperator = boost::any_cast<int>(options.at("epistemicOperator"));
+	if (epistemicOperator == 1) {
+		// Retrieve aboutAgentIRI and confidence
+		KB_INFO("[KnowRob] aboutAgentIRI.");
```
Comment (daniel86) [inline src/integration/InterfaceUtils.cpp]: seems like debug output that should be removed

**Review by daniel86 — CHANGES_REQUESTED** [review]
Looks good, overall just a few small comments. Please run auto formatting

**Comment by sasjonge** [comment]
I fixed all your comments. The identation seems also to be not completly broken anymore after i autoformatted the files.

**Review by daniel86 — APPROVED** [review]

**Pull request merged**